e2e/testenv/infra/cmd/setup.go

```diff
@@ -67,39 +76,52 @@ Parallelism is used when possible to save time.`,
 			fail(fmt.Errorf("create Helm registry client: %v", err))
 		}
 
+		startTime := time.Now()
 		externalIP := os.Getenv("external_ip")
-		if externalIP == "" {
+		for externalIP == "" {
+			if time.Now().After(startTime.Add(timeoutDuration)) {
+				fail(fmt.Errorf("timed out waiting for external IP"))
+			}
+
 			externalIP, err = k.Get("service", "zot-service", "-o", "jsonpath={.status.loadBalancer.ingress[0].ip}")
 			if err != nil {
 				fail(fmt.Errorf("get external Zot service IP: %v", err))
 			}
+			time.Sleep(200 * time.Millisecond)
 		}
```

Comment:
Can we extract this to a function? Also, I think using `context` for timeout will make it cleaner as well.

Reply:
I'll take care of it.